scripts/deps/update-example-versions.js

```diff
@@ -0,0 +1,66 @@
+import path from 'node:path';
+import fs from 'node:fs/promises';
+import { globby as glob } from 'globby';
+
+/*
+  This file updates the dependencies' versions in `examples/*` to match the workspace packages' versions.
+  This should be run after `changeset version` so the release PR updates all the versions together.
+*/
+
+const rootUrl = new URL('../..', import.meta.url);
+const rootPackageJson = JSON.parse(await fs.readFile(new URL('./package.json', rootUrl), 'utf-8'));
+
+// get all workspace package name to versions
+/** @type {Map<string, string>} */
+const packageToVersions = new Map();
+
+// Changeset detects workspace packages to publish via `workspaces` in package.json.
+// Although this conflicts with the `pnpm-workspace.yaml` config, it's easier to configure what gets
+// published through this field, so this file also respects this field when updating the versions.
+const workspaceDirs = await glob(rootPackageJson.workspaces, {
+	onlyDirectories: true,
+	cwd: rootUrl,
+});
+for (const workspaceDir of workspaceDirs) {
+	const packageJsonPath = path.join(workspaceDir, './package.json');
+	const packageJson = await tryParsePackageJson(packageJsonPath);
+	if (!packageJson) continue;
+
+	packageToVersions.set(packageJson.name, packageJson.version);
+}
+
+// Update all examples' package.json
+const exampleDirs = await glob('examples/*', {
+	onlyDirectories: true,
+	cwd: rootUrl,
+});
+for (const exampleDir of exampleDirs) {
+	const packageJsonPath = path.join(exampleDir, './package.json');
+	const packageJson = await tryParsePackageJson(packageJsonPath);
+	if (!packageJson) continue;
+
+	// Update dependencies
+	for (const depName of Object.keys(packageJson.dependencies ?? [])) {
+		if (packageToVersions.has(depName)) {
+			packageJson.dependencies[depName] = `^${packageToVersions.get(depName)}`;
```

Comment:
nit: should we just pin `~` - bug fixes?

Personally, I would pin the versions without ranges. It's known that there are libraries that break the CIs with patches.

Reply:
Worth bearing in mind we’re bumping `astro` or `@astrojs/*` integrations in our example templates here. If we break CI with patches, that’s probably on us, no?

Reply:
I think `^` should be good here as it's what we used to use too. Since these are our packages that we're bumping, I don't think we're making breaking changes in minors or patches that needs this restriction.